hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/HBaseInterClusterReplicationEndpoint.java

```diff
@@ -475,7 +476,7 @@ public boolean replicate(ReplicateContext replicateContext) {
     }
 
     List<List<Entry>> batches = createBatches(replicateContext.getEntries());
-    while (this.isRunning() && !exec.isShutdown()) {
+    while (this.isRunning() && !exec.isShutdown() && !Thread.currentThread().isInterrupted()) {
```

Comment:
This means we eat the interrupted exception at some places? Can we propagate it to the upper layer?

Reply:
Yes, InterruptedException catched in method sleepForRetries, but no further processing. The upper layer, you said, means ReplicationSourceShipper? We just keep the interrupted status of the current thread, ReplicationSourceShipper.isActive() can sense it.

Reply:
Then I think we should reset the running or shutdown flag in sleepForRetries when we get InterruptedException? Then here we know that we should quit?

Reply:
How about using `ReplicationSource.isSourceActive()` instead of `!Thread.currentThread().isInterrupted()` ? HBaseReplicationEndpoint manages lifecycle by extent `AbstractService`, adding a flag in HBaseInterClusterReplicationEndpoint is a bit redundant?